src/microbe_stage/CompoundCloudSystem.cs

```diff
@@ -1,210 +1,182 @@
+            {
+                var cloudNode = cloudScene.Instantiate<Node>();
 
-            var cloud1 = allCloudCompounds[startOffset + 0].ID;
-            var cloud2 = Compound.Invalid;
-            var cloud3 = Compound.Invalid;
-            var cloud4 = Compound.Invalid;
+                if (cloudNode == null)
+                {
+                    GD.PrintErr($"Failed to create CompoundCloudPlane instance {i}");
+                    return;
+                }
 
-            if (startOffset + 1 < allCloudCompounds.Count)
-                cloud2 = allCloudCompounds[startOffset + 1].ID;
+                AddChild(cloudNode);
+                cloudInstances.Add(cloudNode);
 
-            if (startOffset + 2 < allCloudCompounds.Count)
-                cloud3 = allCloudCompounds[startOffset + 2].ID;
+                int startOffset = i * Constants.CLOUDS_IN_ONE;
+                int cloud1 = (startOffset < allCloudCompounds.Count) ? GetCompoundID(allCloudCompounds[startOffset]) : -1;
+                int cloud2 = (startOffset + 1 < allCloudCompounds.Count) ? GetCompoundID(allCloudCompounds[startOffset + 1]) : -1;
+                int cloud3 = (startOffset + 2 < allCloudCompounds.Count) ? GetCompoundID(allCloudCompounds[startOffset + 2]) : -1;
+                int cloud4 = (startOffset + 3 < allCloudCompounds.Count) ? GetCompoundID(allCloudCompounds[startOffset + 3]) : -1;
 
-            if (startOffset + 3 < allCloudCompounds.Count)
-                cloud4 = allCloudCompounds[startOffset + 3].ID;
+                cloudNode.Call("init", -i - 1, cloud1, cloud2, cloud3, cloud4);
+                cloudNode.Set("position", Vector3.Zero);
+            }
 
-            clouds[i].Init(fluidSystem, renderPriority, cloud1, cloud2, cloud3, cloud4);
-            --renderPriority;
-            clouds[i].Position = new Vector3(0, 0, 0);
+            GD.Print("CompoundCloudSystem initialization completed successfully");
+        }
+        catch (Exception e)
+        {
+            GD.PrintErr($"Error in CompoundCloudSystem.Init: {e.Message}\n{e.StackTrace}");
         }
     }
 
     public override void _Process(double delta)
     {
-        elapsed += delta;
+        if (isExiting)
+            return;
 
-        // Limit the rate at which the clouds are processed as they
-        // are a major performance sink
-        if (elapsed >= Settings.Instance.CloudUpdateInterval)
+        try
         {
-            UpdateCloudContents((float)elapsed);
-            elapsed = 0;
+            elapsed += delta;
+
+            if (elapsed >= Settings.Instance.CloudUpdateInterval)
+            {
+                elapsed = 0;
+                foreach (var cloudInstance in cloudInstances)
+                {
+                    cloudInstance.Call("update_texture");
+                }
+            }
         }
+        catch (Exception e)
+        {
+            GD.PrintErr($"Error in CompoundCloudSystem._Process: {e.Message}\n{e.StackTrace}");
```

Comment:
I'm pretty sure there's no way anything here would cause an exception... because Godot Call will just print an error and not throw anything.